graphdatascience/procedure_surface/api/catalog/node_properties_endpoints.py

```diff
@@ -0,0 +1,169 @@
+        self,
+        G: Graph,
+        node_properties: NodePropertySpec,
+        *,
+        node_labels: Optional[List[str]] = None,
+        concurrency: Optional[Any] = None,
+        write_concurrency: Optional[Any] = None,
+        sudo: Optional[bool] = None,
+        log_progress: Optional[bool] = None,
+        username: Optional[str] = None,
+        job_id: Optional[Any] = None,
+    ) -> NodePropertiesWriteResult:
+        """
+        Writes the specified node properties from the graph to the database.
+
+        Parameters
+        ----------
+        G : Graph
+            The graph to write node properties from
+        node_properties : NodePropertySpec
+            The node properties to stream
+        node_labels : Optional[List[str]], default=None
+            Filter by node labels
+        concurrency : Optional[Any], default=None
+            The number of concurrent threads
+        write_concurrency : Optional[Any], default=None
+            The number of concurrent threads used for writing
+        sudo : Optional[bool], default=None
+            Override memory estimation limits
+        log_progress : Optional[bool], default=None
+            Whether to log progress
+        username : Optional[str], default=None
+            The username to attribute the procedure run to
+        job_id : Optional[Any], default=None
+            An identifier for the job
+        Returns
+        -------
+        NodePropertiesWriteResult
+            Execution metrics and statistics
+        """
+        pass
+
+    @abstractmethod
+    def drop(
+        self,
+        G: Graph,
+        node_properties: List[str],
+        *,
+        fail_if_missing: Optional[bool] = None,
+        concurrency: Optional[Any] = None,
+        sudo: Optional[bool] = None,
+        log_progress: Optional[bool] = None,
+        username: Optional[str] = None,
+        job_id: Optional[Any] = None,
+    ) -> NodePropertiesDropResult:
+        """
+        Drops the specified node properties from the graph.
+
+        Parameters
+        ----------
+        G : Graph
+            The graph to drop node properties from
+        node_properties : List[str]
+            The node properties to drop
+        fail_if_missing: Optional[bool] = None,
+            Whether to fail if any of the node properties are missing
+        concurrency : Optional[Any], default=None
+            The number of concurrent threads
+        sudo : Optional[bool], default=None
+            Override memory estimation limits
+        log_progress : Optional[bool], default=None
+            Whether to log progress
+        username : Optional[str], default=None
+            The username to attribute the procedure run to
+        job_id : Optional[Any], default=None
+            An identifier for the job
+        Returns
+        -------
+        NodePropertiesDropResult
+            Execution metrics and statistics
+        """
+        pass
+
+
+class NodePropertySpec:
```

Comment:
i would suggest to use a union type for the user to instantiate. internally we could transform it to the spec though.

union type such as: `Union[str, List[str], dict[str, str]]`